### PR TITLE
fix: use groups and username from tokenreview auth step in authconfig, fixes RHOAIENG-16025

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description

Use groups and username from tokenreview auth step in authconfig
Fixes RHOAIENG-16025

Requires merging and releasing Authorino fix first at https://github.com/Kuadrant/authorino/issues/506
Fix for which is in the PR https://github.com/Kuadrant/authorino/pull/507

## Merge criteria:

- [X] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: config/crd/bases/modelregistry.opendatahub.io_modelregistries.yaml [hunk 1]
```diff
@@ -45,30 +45,38 @@ spec:
...
+              In CamelCase.
+              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
             type: string
           metadata:
             type: object
           spec:
-            description: ModelRegistrySpec defines the desired state of ModelRegistry.
+            description: |-
+              ModelRegistrySpec defines the desired state of ModelRegistry.
               One of `postgres` or `mysql` database configurations MUST be provided!
```
Comment (lugi0) [inline config/crd/bases/modelregistry.opendatahub.io_modelregistries.yaml:67]: We've verified that MariaDB is supported as well, should we update this?

File: internal/controller/config/templates/istio/authconfig.yaml.tmpl [hunk 2]
```diff
@@ -35,9 +35,9 @@ spec:
     k8s-rbac:
       kubernetesSubjectAccessReview:
         user:
-          selector: auth.identity.metadata.annotations.userid
-        groups:
-          - {{.Name}}-users
+          selector: auth.identity.user.username
+        authorizationGroups:
+          selector: auth.identity.user.groups
```
Comment (lugi0) [inline internal/controller/config/templates/istio/authconfig.yaml.tmpl:40]: Does this behave as a `OR` or as an `AND`? i.e. will a user who is allowed but is not part of an allowed group be considered allowed?

**Review by Al-Pragliola — APPROVED** [review]
lgtm

File: config/crd/bases/modelregistry.opendatahub.io_modelregistries.yaml (hunk 1 above)
Comment (dhirajsb) [reply config/crd/bases/modelregistry.opendatahub.io_modelregistries.yaml:67]: afair, MariaDB is not recommended for production by the DSP team, so I'm inclined to leave this as is.

File: internal/controller/config/templates/istio/authconfig.yaml.tmpl (hunk 2 above)
Comment (dhirajsb) [reply internal/controller/config/templates/istio/authconfig.yaml.tmpl:40]: The selector grabs the list of all groups that the user is part of from the K8s TokenReview. So, it actually dynamically tests all user/serviceaccount groups rolebdindings as well as the user/SA's own rolebindings. 

So, users can either bind to individual accounts or groups.

**Comment by dhirajsb** [comment]
@Al-Pragliola once @lugi0 marks his questions as resolved, we can merge this PR.

**Review by lugi0 — APPROVED** [review]
lgtm!

**Pull request merged**